wallet/src/account/output_cache/mod.rs

```diff
@@ -239,11 +239,17 @@ impl OutputCache {
             .collect()
     }
 
-    pub fn abandon_transaction(&mut self, tx_id: Id<Transaction>) -> WalletResult<()> {
+    pub fn abandon_transaction(
+        &mut self,
+        tx_id: Id<Transaction>,
+    ) -> WalletResult<Vec<Id<Transaction>>> {
+        let mut all_abandoned = Vec::new();
         let mut to_abandon = BTreeSet::new();
         to_abandon.insert(OutPointSourceId::from(tx_id));
```

Comment:
Unrelated to this PR. I think the initialisation can be simplified roughly like so:
```rust
let mut to_abandon = BTreeSet::from_iter([OutPointSourceId::from(tx_id)]);
```

Reply:
fixed